File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
    int answer = 0;
    
    sort(citations.begin(), citations.end());
    
    int left = citations[0];
    int right = citations[citations.size() - 1];
    
    for(int i = 1; i <= right; i++){
        
        int cnt = 0;
        for(auto a : citations){
            if(a >= i)
                cnt++;
        }
        
        answer = max(answer, min(cnt, i));
    }   
    return answer;
}

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        
        Arrays.sort(citations);

        int start = 0;
        int end = citations[citations.length - 1];
        
        for(int i = start; i <= end; i++){
            
            int cnt = 0;
            for(int now : citations){
                if(now >= i)
                    cnt++;
            }
            
            int hidx = Math.min(cnt, i);
            answer = Math.max(answer, hidx);
        }
        
        return answer;
    }
}